src/main/java/lotto/domain/WinningInfo.java

```diff
@@ -0,0 +1,63 @@
+package lotto.domain;
+
+public enum WinningInfo {
+    FAIL(0, false, 0),
+    FIFTH(3, false, 5000),
+    FOURTH(4, false, 50000),
+    THIRD(5, false, 1500000),
+    SECOND(5, true, 30000000),
+    FIRST(6, false, 2000000000);
+
+    private int winningCount;
+    private boolean hasBonus;
+    private int winningPrice;
+
+    WinningInfo(int winningCount, boolean hasBonus, int winningPrice) {
+        this.winningCount = winningCount;
+        this.hasBonus = hasBonus;
+        this.winningPrice = winningPrice;
+    }
+
+    public static WinningInfo valueOf(int winningCount, boolean hasBonus) {
```

Comment:
stream을 활용해서 좀더 간결하게 써보면 어떨까요?
```suggestion
        final WinningInfo winning = Arrays.stream(values())
                .filter(x -> x.winningCount == winningCount)
                .findFirst()
                .orElse(FAIL);

        if (hasBonus && (winning == THIRD || winning == SECOND)) {
            return SECOND;
        }
        return winning;
```